src/main/proto/com/google/api/codegen/grpc/service_config.proto

```diff
@@ -0,0 +1,308 @@
+// A ServiceConfig is supplied when a service is deployed. It mostly contains
+// parameters for how clients that connect to the service should behave (for
+// example, the load balancing policy to use to pick between service replicas).
+//
+// The configuration options provided here act as overrides to automatically
+// chosen option values. Service owners should be conservative in specifying
+// options as the system is likely to choose better values for these options in
+// the vast majority of cases. In other words, please specify a configuration
+// option only if you really have to, and avoid copy-paste inclusion of configs.
+//
+// DISCLAIMER: This is a copy of the ServiceConfig proto found in
+// https://github.com/grpc/grpc-proto modified only slightly to integrate with
+// how proto dependencies are generated in this repo. If changes are made to the
+// original proto they must be copied here.
+
+syntax = "proto3";
+
+package com.google.api.codegen.grpc;
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/wrappers.proto";
+import "google/rpc/code.proto";
+
+option java_package = "com.google.api.codegen.grpc";
+option java_multiple_files = true;
+option java_outer_classname = "ServiceConfigProto";
+
+// Configuration for a method.
+message MethodConfig {
+  // The names of the methods to which this configuration applies. There must
+  // be at least one name. Each name entry must be unique across the entire
+  // ClientConfig. If the 'method' field is empty, then this MethodConfig
+  // specifies the defaults for all methods for the specified service.
+  //
+  // For example, let's say that the service config contains the following
+  // MethodConfig entries:
+  //
+  // method_config { name { service: "MyService" } ... }
+  // method_config { name { service: "MyService" method: "Foo" } ... }
+  //
+  // For a request for MyService/Foo, we will use the second entry, because it
+  // exactly matches the service and method name.
+  // For a request for MyService/Bar, we will use the first entry, because it
+  // provides the default for all methods of MyService.
+  message Name {
```

Comment:
Is it possible to pull this out to the top level? (to be consistent with gapic config definitions, and to make import statements easier)

Also is it possible to change the name of this message `Name`? Another class called `Name` is defined in this repo and is fairly commonly used. Changing this to some other name would save the potential trouble of having to refer to one of them with fully qualified name when used in the same file.

Reply:
Your requests are very reasonable and I would normally implement them, but I don't want to change this proto at all because it is a direct copy of another proto (as mentioned in this file's header comment). If we change it, it diverts from the original (which shouldnt change very much at all, but still).

Reply:
That makes sense.